### PR TITLE
Dodging effects

### PR DESCRIPTION
**Pull request by Helfull**
merging minor fixes first please :)

### CODE REVIEW

File: jack_bot.py [hunk 1]
```diff
@@ -137,69 +140,7 @@ def __init__(self, debug=False):
         self.locations = []
         self.ordered_expansions = []
         self.building_positions = []
-        self.close_enemies_to_base = False
-        self.close_enemy_production = False
-        self.floating_buildings_bm = False
-        self.counter_attack_vs_flying = False
```
Comment (Matuiss2) [inline jack_bot.py]: this is missing on the refactoring

File: jack_bot.py [hunk 2]
```diff
@@ -232,27 +174,6 @@ def set_game_step(self):
             self.prepare_expansions()
             self.split_workers()
 
-        if self.ground_enemies:
```
Comment (Matuiss2) [inline jack_bot.py]: here too is missing on the new one

File: profile.bat [hunk 3]
```diff
@@ -0,0 +1,10 @@
+@ECHO off
```
Comment (Matuiss2) [inline profile.bat]: this file is needed?

File: actions/micro.py [hunk 4]
```diff
@@ -1,12 +1,32 @@
...
 class Micro:
     """Group all helpers, for unit control and targeting here"""
 
+    def dodge_effects(self, unit: Unit) -> bool:
+        """Dodge any effects"""
+        if self.ai.state.effects:
+            for effect in self.ai.state.effects:
+                effect_data = self.ai._game_data.effects[effect.id]
+                danger_zone = effect_data.radius + unit.radius + .1
+                clostest_effect_position_to_unit = unit.position.closest(effect.positions)
```
Comment (tweakimp) [inline actions/micro.py]: Rename variable. Suggestion: Kloßtest_effect...

File: actions/micro.py [hunk 5]
```diff
@@ -1,12 +1,32 @@
...
+
+                # Are we in the danger zone
+                if unit.position.distance_to_point2(clostest_effect_position_to_unit) < danger_zone:
+                    move_away = -1 * danger_zone
+                    neighbors8_of_unit = list(unit.position.neighbors8)
+                    center_of_effect = Point2.center(effect.positions)
+                    clostest_neighbor_to_effect = center_of_effect.furthest(neighbors8_of_unit)
```
Comment (tweakimp) [inline actions/micro.py]: Rename variable. Suggestion: Furthest...

File: jack_bot.py (hunk 1 above)
Comment (Helfull) [reply jack_bot.py]: please review these changes in here https://github.com/Matuiss2/Greedy_bot/pull/148

File: jack_bot.py (hunk 2 above)
Comment (Helfull) [reply jack_bot.py]: please review https://github.com/Matuiss2/Greedy_bot/pull/148 first, once minor-fixes is merged, this PR will be less of code and only contains the changes to dodging

File: profile.bat (hunk 3 above)
Comment (Helfull) [reply profile.bat]: i would keep this file so everyone can profile the bot without a lot of knowledge

**Comment by Helfull** [comment]
DONE MERGE

**Pull request merged**